### DISCUSSION TITLE
Filter option to be hidden like column

### DISCUSSION THREAD

**Discussion by dgillier** [rappasoft/laravel-livewire-tables#1158] [Feature Requests]
It will be nice to have a ->hideIf() function for filter, for exemple to hide filters depending on user roles ?

Thanks for considering,
Denis :)

**Comment by lrljoe** [comment]
There is the hiddenFromAll() capability, but can't remember offhand if that has a callback option, don't think it does.

Shouldn't be too painful to add, good call.

In the meantime you could just call that function on the relevant filter in your configure function, let me know if you need that clarifying.

**Reply by lrljoe** [reply]
Initial tests look okay, but need to play a bit further before I'm comfortable adding this in.  I'd probably anticipate it being in the version after next.

**Reply by lrljoe** [reply]
Probably 2.15 or 2.16 if I'm being honest, it seems to work, I just don't have time this week to write the tests for it, and I'm not adding any features without the tests, as Anthony has done a great job on the tests.

**Reply by lrljoe** [reply]
Linked to a Feature Request issue to allow for tracking in the project
